slackbot/slack.go

```diff
@@ -535,25 +531,34 @@ func HerokuPing() {
 	}
 }
 
-func (robot Robot) Run() {
+func MessageWorker(robot *Robot) {
+	for {
```

Comment:
You should use a range here, so this worker stops when the channel is closed (even if you don't care about that, it's better)